server/src/main.js

```diff
@@ -26,64 +31,104 @@ const { User, Device } = models;
 
 const push = getPushEmitters({ models });
 
-const handlers = getHandlers({ models, creators, push });
+const handlers = getHandlers({ models, creators, push, pubsub });
 const resolvers = getResolvers(handlers);
 const schema = getSchema(resolvers);
 
-
 const app = express();
 
 app.use(cors());
 
-// `context` must be an object and can't be undefined when using connectors
+// Take incoming JTW and resolve user and device and return it. If not passed then use DEFAULT
+// user and device.
+const postJTWAuth = ({ id, device }) => new Promise((resolve) => {
+  let returnUser;
+  let returnDevice;
+  if (id && device) {
+    returnUser = User.findById(id);
+    returnDevice = Device.findOne({
+      where: {
+        userId: id,
+        uuid: device,
+      },
+    });
+  } else {
+    // If we didnt allow anon access we would reject the promise here
+    returnUser = User.findById(DEFAULT_USER_ID);
+    returnDevice = Device.findOne({
+      where: {
+        userId: DEFAULT_USER_ID,
+        uuid: DEFAULT_DEVICE_UUID,
+      },
+    });
+  }
+
+  resolve({ user: returnUser, device: returnDevice });
```

Comment:
you're synchronously returning the promise so there's no need for a promise. If you want to return a promise so that you can make this function async in the future without breaking callers then use `return Promise.resolve({ ... });` at the end instead.

```
const postJWTAuth = ({ id, device }) => {
  ...

  return { user: returnUser, device, returnDevice };
}
```